src/Service/Catalog/Product.php

```diff
@@ -145,19 +157,49 @@ private function shouldSynchronize(MageProduct $product)
      */
     public function update(MageProduct $product): void
     {
-        // TODO remove, when configurable product sync is implemented
-        if ($product->getTypeId() === TypeConfigurable::TYPE_CODE) {
-            throw new UnsupportedProductType();
-        }
-
         $this->shouldSynchronize($product);
-        $list = $this->productHelper->getProductDtoMap($product, ProductHelper::ACTION_UPDATE);
-        foreach ($list as $storeGroup => $dto) {
-            $this->utility->initSdk()->getCatalogService()->updateProduct((string) $product->getId(), $dto);
+        $updates = $this->aggregateProductIdsForUpdate($product);
+
+        foreach ($updates as $productId => $dto) {
+            $this->utility->initSdk()->getCatalogService()->updateProduct((string) $productId, $dto);
```

Comment:
make sure this is true on the `aggregateProductIdsForUpdate` end
```suggestion
            $this->utility->initSdk()->getCatalogService()->updateProduct($productId, $dto);
```